activemodel/test/cases/validations_test.rb

```diff
@@ -159,6 +160,15 @@ def test_validate_block_with_params
     assert_equal ["will never be valid"], t.errors["title"]
   end
 
+  def test_validates_with_array_condition
+    assert_nothing_raised do
+      Topic.with_options(on: :create, if: %i[persisted?]) do
```

Comment:
Let's change this test to not use `with_options` and instead set the value of the `:if` option to a variable and assert that the value in the variable is the same after we call `validate`.

Something like this.

```
def test_validates_with_array_condition_does_not_mutate_the_array
  opts = []
  validate { }, if: opts
  assert_empty opts
end
```